src/timer/TimingInterface.ts

```diff
@@ -0,0 +1,148 @@
+class TimingInterface {
+    initialTime: number;
+    currentTime: number;
+    rate:number;
+    timerId: number;
+
+    constructor (duration: number, rate: number = 1) {
+        this.initialTime = duration;
+        this.currentTime = duration;
+        this.rate = rate;
+    }
+
+    getCurrentTime() {
```

Comment:
typescript has default setters and getters.
You can do the following get currentTime(){
return this.currentTIme
}

https://stackoverflow.com/questions/12827266/get-and-set-in-typescript